llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp

```diff
@@ -721,6 +721,8 @@ DecodeStatus RISCVDisassembler::getInstruction32(MCInst &MI, uint64_t &Size,
                         "Qualcomm uC Conditional Move custom opcode table");
   TRY_TO_DECODE_FEATURE(RISCV::FeatureVendorXqciint, DecoderTableXqciint32,
                         "Qualcomm uC Interrupts custom opcode table");
+  TRY_TO_DECODE_FEATURE(RISCV::FeatureVendorXRivosVizip, DecoderTableXRivos32,
```

Comment:
I noticed that many of the existing extensions have one decode table per extension.  I thought the idea was we had one decode table *per vendor* unless and until that vendor had an encoding conflict in their own extension sets?

Reply:
I'm glad you've pointed this out, as it did seem like we are trying to side-step the encoding conflict checking that LLVM is capable of, as well as the fact that the decode tables will also check specific predicates for specific instructions.

I will look at merging all the Qualcomm decode tables, given that all the Xqci instructions are intended not to conflict with each other's encodings. I am less clear on which other vendor extensions should or should not conflict with each other.

I think the `TRY_TO_DECODE_FEATURE(...` encouraged one decode table per feature - it's relatively more difficult to write an equivalent for when you have any of a few different features - but we can solve this with yet more macros of course.

Reply:
JFYI, repeating the existing macro per extension with the same table name works just fine.  You could also do other structures, but the direct migration is pretty easy - just change the table names.